tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -543,6 +545,16 @@ spv::LoopControlMask SPIRVEmitter::translateLoopAttribute(const Attr &attr) {
   return spv::LoopControlMask::MaskNone;
 }
 
+void SPIRVEmitter::doDiscardStmt(const DiscardStmt *discardStmt) {
+  assert(!theBuilder.isCurrentBasicBlockTerminated());
+  theBuilder.createKill();
+  if (!isLastStmtBeforeControlFlowBranching(astContext, discardStmt)) {
```

Comment:
The logic for break statements is a bit different. I'll consider it for another CL.